test/e2e/e2e_test.go

```diff
@@ -237,6 +239,105 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("verify userAccount is not deleted if namespace is not deleted", func(t *testing.T) {
+		laraSignUp, _ := NewSignupRequest(t, hostAwait, memberAwait, memberAwait2).
+			Username("laracroft").
+			Email("laracroft@redhat.com").
+			ManuallyApprove().
+			EnsureMUR().
+			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+			Execute().Resources()
+
+		require.Equal(t, "laracroft", laraSignUp.Status.CompliantUsername)
+
+		laraUserName := "laracroft"
+		userNamespace := "laracroft-dev"
+		podName := "test-useraccount-delete-1"
+
+		//Create a pod with finalizer in user's namespace, which will block the deletion of namespace.
+		memberPod := corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: userNamespace,
+				Finalizers: []string{
+					"test/finalizer.toolchain.e2e.tests",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{{
+					Name:  "test",
+					Image: "busybox",
+				}},
+			},
+		}
+		err = memberAwait.Client.Create(context.TODO(), &memberPod)
+		require.NoError(t, err)
+		pod, err := memberAwait.WaitForPod(userNamespace, podName)
+		require.NoError(t, err)
+		require.NotEmpty(t, pod)
+
+		deletePolicy := metav1.DeletePropagationForeground
+		deleteOpts := &client.DeleteOptions{
+			PropagationPolicy: &deletePolicy,
+		}
+		// now delete userSignup but nothing should be deleted yet
+		err = hostAwait.Client.Delete(context.TODO(), laraSignUp, deleteOpts)
+		require.NoError(t, err)
+
+		// Check that namespace is not deleted and is in Terminating state after 10sec
+		_, err = memberAwait.WithRetryOptions(wait.TimeoutOption(time.Second * 10)).WaitForNamespaceInTerminating(userNamespace)
+		require.NoError(t, err)
+
+		// Check all corresponding resources are blocked by the terminating ns and have expected conditions
+		NSTemplateSetCondition := toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.ConditionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NSTemplateSetTerminatingFailedReason,
+			Message: "user namespace laracroft-dev deletion was triggered but is not complete yet, something could be blocking ns deletion",
+		}
```

Comment:
Should we move this to a function in testsupport/conditions.go too? It can just accept a parameter for the namespace that we can set to `laracroft-dev`?